spec/lib/approval_request_listener_spec.rb

```diff
@@ -0,0 +1,117 @@
+describe ApprovalRequestListener do
+  let(:client)               { double(:client) }
+  let(:event)                { ApprovalRequestListener::EVENT_REQUEST_FINISHED }
+  let(:approval_request_ref) { "0" }
+  let(:payload)              { {"request_id" => request_id, "decision" => decision, "reason" => reason } }
+  let(:request_id)           { "1" }
+
+  describe "#subscribe_to_approval_updates" do
+    let(:request) do
+      { :headers => { 'x-rh-identity' => encoded_user_hash }, :original_url => 'whatever' }
+    end
+
+    around do |example|
+      ManageIQ::API::Common::Request.with_request(request) { example.call }
+    end
+
+    let(:messages) { [ManageIQ::Messaging::ReceivedMessage.new(nil, event, payload, nil)] }
+    let!(:approval_request) do
+      ApprovalRequest.create!(
+        :approval_request_ref => approval_request_ref,
+        :workflow_ref         => "2323",
+        :order_item_id        => "3434"
+      )
+    end
+
+    before do
+      allow(ManageIQ::Messaging::Client).to receive(:open).with(
+        :protocol   => :Kafka,
+        :client_ref => ApprovalRequestListener::CLIENT_AND_GROUP_REF,
+        :group_ref  => ApprovalRequestListener::CLIENT_AND_GROUP_REF
+      ).and_return(client)
+      allow(client).to receive(:subscribe_topic).with(
+        :service   => ApprovalRequestListener::SERVICE_NAME,
+        :max_bytes => 500_000
+      ).and_yield(messages)
+      allow(client).to receive(:close)
+    end
+
+    context "when the approval request is not findable" do
+      context "when the request_id is anything else" do
+        let(:request_id) { "10" }
+        let(:payload)    { {"request_id" => request_id } }
```

Comment:
`request_id` can just be inlined to `"request_id" => "10"` and remove the `let` above since we aren't using it anywhere for expectations.